src/Common/tests/Tests/System/IO/PathInternal.Windows.Tests.cs

```diff
@@ -141,5 +140,102 @@ public void AreRootsEqual(string first, string second, StringComparison comparis
         {
             Assert.Equal(expected, PathInternal.AreRootsEqual(first, second, comparisonType));
         }
+
+        public static TheoryData<string, int> GetRootLength_Data => new TheoryData<string, int>
+        {
+            { @"C:\git\corefx", 3},
+            { @"C:\git\.\", 3},
+            { @"C:\git\..\", 3},
+            { @"C:\git\..\..\", 3},
+            { @"C:\..\", 3},
+            { @"C:\", 3},
+            { @"C:", 2},
+            { @"C:git\corefx", 2},
+            { @"C:git\.\", 2},
+            { @"C:git\..\", 2},
+            { @"C:..\", 2},
+        };
+
+        public static TheoryData<string, int> GetRootLengthRooted => new TheoryData<string, int>
+        {
+            { @"\tmp", 1},
+            { @"\.", 1},
+            { @"\..", 1},
+            { @"\tmp\..\..", 1},
+            { @"\\", 2},
+            { @"\\\", 3},
+        };
+
+        [Theory,
+            MemberData(nameof(GetRootLength_Data))]
+        public void GetRootLength(string path, int length)
+        {
+            Assert.Equal(length, PathInternal.GetRootLength(path));
+            Assert.Equal(length + PathInternal.ExtendedPathPrefix.Length, PathInternal.GetRootLength(@"\\?\" + path));
+            Assert.Equal(length + PathInternal.ExtendedPathPrefix.Length, PathInternal.GetRootLength(@"\\.\" + path));
+        }
+
+        [Theory,
+    MemberData(nameof(GetRootLengthRooted))]
+        public void GetRootLength_Rooted(string path, int length)
+        {
+            Assert.Equal(length, PathInternal.GetRootLength(path));
+        }
+
+        public static TheoryData<string, int> GetRootLength_UNCData => new TheoryData<string, int>
+        {
+            { @"Server\Share\git\corefx", 13},
+            { @"Server\Share\git\.\", 13},
+            { @"Server\Share\git\..\", 13},
+            { @"Server\Share\git\..\..\", 13},
+            { @"Server\Share\..\", 13},
+            { @"Server\Share\", 13},
+            { @"Server\Share", 12},
+            { @"\\Server\Share", 14},
+            //{ @"\\Server\\Share", 15},
```

Comment:
Don't really care what we get for `\\\Server\Share`, `\\\\Server\\Share`, etc. It's good to test, but we're ok with a mess as Windows won't handle the paths well. You can use that as a metric- if GetFullPathName doesn't fix the separators correctly we can ignore it.

Bottom line is that up to the end of the "volume" I expect the # of separators provided to be correct. Outside of that, don't make the code any more complicated to deal with it.  (E.g. up to `C:` `C:\` `\\server\share`, `\\?\UNC\Server\Share` etc we should get the right number of separators.) Collapsing separators past that point is meant to be friendly to path concatenation.